src/VisualStudio/Core/Def/Implementation/TaskList/ExternalErrorDiagnosticUpdateSource.cs

```diff
@@ -799,21 +801,25 @@ static bool IsDocumentLevelDiagnostic(DiagnosticData diagnosticData)
                 }
             }
 
-            private bool IsSupportedLiveDiagnosticId(Project project, string id)
-                => GetOrCreateSupportedLiveDiagnostics(project).Contains(id);
+            private async ValueTask<bool> IsSupportedLiveDiagnosticIdAsync(Project project, string id, CancellationToken cancellationToken)
+            {
+                var supportedLiveDiagnostics = await GetOrCreateSupportedLiveDiagnosticsAsync(project, cancellationToken).ConfigureAwait(false);
+                return supportedLiveDiagnostics.Contains(id);
+            }
 
-            private ImmutableHashSet<string> GetOrCreateSupportedLiveDiagnostics(Project project)
+            private async ValueTask<ImmutableHashSet<string>> GetOrCreateSupportedLiveDiagnosticsAsync(Project project, CancellationToken cancellationToken)
             {
+                var analysisScope = await _owner._analysisScopeService.GetAnalysisScopeAsync(project, cancellationToken).ConfigureAwait(false);
+                var fullSolutionAnalysis = analysisScope == BackgroundAnalysisScope.FullSolution;
+                if (!project.SupportsCompilation || fullSolutionAnalysis)
+                {
+                    return GetOrCreateSupportedDiagnosticIds(project.Id);
+                }
+
                 return GetOrCreateDiagnosticIds(project.Id, _liveDiagnosticIdMap, ComputeSupportedLiveDiagnosticIds);
```

Comment:
@mavasani prior to this change, `_liveDiagnosticIdMap` was getting different values depending on the current analysis scope configuration. This meant the solution crawler would fail to use the correct new values until the IDE is restarted after a change to analysis scope in Tools&rarr;Options.